index.js

```diff
@@ -14,10 +14,11 @@ function gulpPugLinter () {
    * @description Checks the given file for lint errors
    * @param {Object} file File chunk
    * @param {String} file.path File path
+   * @param {Buffer} file.contents Contents of the file.
    * @returns {Array} List of error messages
    */
   function checkFile (file) {
-    return linter.checkFile(file.path)
+    return linter.checkString(file.contents.toString('utf-8'), file.path)
```

Comment:
The difference between [`linter.checkFile()`](https://github.com/pugjs/pug-lint/blob/master/lib/linter.js#L42-L49) and [`linter.checkString()`](https://github.com/pugjs/pug-lint/blob/master/lib/linter.js#L67-L73) is that the former makes sure that the files are not configured to be ignored. Please [read through line #44](https://github.com/pugjs/pug-lint/blob/master/lib/linter.js#L44) to see it for yourself.

You can see how this changes breaks the functionality of the plugin. Simply add `"excludeFiles": ["..."]` to your test configuration object, and notice how with these changes, the specified files are no longer being ignored. Because `excludedFiles` is a [documented function](https://github.com/pugjs/pug-lint#excludefiles) of [_pug-lint_](https://github.com/pugjs/pug-lint), I cannot approve this change.

Moreover, it seems like the underlying premise behind reading from the file's contents, as opposed to from the file's path, was to improve the speed of this plugin. However, when running the benchmarks manually (provided all things being equal), here were the actual albeit surprising results:

```
Baseline:
'test' errored after 76 ms
'test' errored after 73 ms
'test' errored after 72 ms
'test' errored after 76 ms

With fix:
'test' errored after 78 ms
'test' errored after 82 ms
'test' errored after 73 ms
'test' errored after 89 ms
```

As you can see, the average execution time for this plugin has increased by 6.25ms.

To benchmark, I've used the same machine with the same test files of`gulp.js`, `.pug-lint.json`, and `test.pug` against the two branches: this one and `develop`. You can read more about the overall setup [in the `CONTRIBUTING#Getting Started` section](https://github.com/ilyakam/gulp-pug-linter/blob/develop/CONTRIBUTING.md#getting-started).